membrane_curvature/surface.py

```diff
@@ -97,11 +97,68 @@ def normalized_grid(grid_z_coordinates, grid_norm_unit):
 
     Returns
     -------
-    Returns nornalized set of z coordinates in grid.
+    Returns normalized set of z coordinates in grid.
 
     """
 
     grid_norm_unit = np.where(grid_norm_unit > 0, grid_norm_unit, np.nan)
     z_normalized = grid_z_coordinates / grid_norm_unit
 
     return z_normalized
+
+
+def interpolation_by_array(array_surface):
+    """
+        Interpolates values contained in array_surface over axi
+
+    Parameters
+    ----------
+
+    array_surface: numpy.ndarray
+        Numpy array of floats of shape (`n_x_bins`, `n_y_bins`)
+
+    Returns
+    -------
+    interpolated_array: np.ndarray
+        Returns interpolated array.
+        Numpy array of shape (`n_x_bins` x `n_y_bins`,)
+
+    """
+
+    # create mask for nans
+    mask_nans = np.isnan(array_surface)
+
+    # index of array_surface
+    index_array = np.arange(array_surface.shape[0])
+
+    # interpolate values in array
+    interpolated_array = np.interp(index_array,
+                                   np.flatnonzero(~mask_nans),
+                                   array_surface[~mask_nans])
+
+    return interpolated_array
+
+
+def surface_interpolation(array_surface):
+    """
+        Calculates interpolation
+
+    Parameters
+    ----------
+
+    array_surface: np.ndarray
+        Numpy array of floats of shape (`n_x_bins`, `n_y_bins`)
```

Comment:
Here and in a few other places: Numpy -> NumPy

Could also just omit the NumPy specification or use `np.array` instead perhaps.